index.js

```diff
@@ -440,19 +486,43 @@ function collectObj (stream, cb) {
   function end () { cb(null, rows) }
 }
 
+// Object, (k, v -> v) -> Object
 function mapObj (obj, fn) {
   Object.keys(obj).forEach(function (key) {
     obj[key] = fn(key, obj[key])
   })
   return obj
 }
 
-function filterObj (obj, fn) {
-  var res = {}
-  Object.keys(obj).forEach(function (key) {
-    if (fn(key, obj[key])) {
-      res[key] = obj[key]
+// KdbPoint -> OsmVersion
+function kdbPointToVersion (pt) {
+  return pt.value.toString('hex')
+}
+
+// OsmDb, OsmVersion -> [OsmId]
+function getWayNodesOfDeletedDoc (osm, version, cb) {
+  getTypeOfDeletedDoc(osm, version, function (err, type) {
+    if (err) return cb(err)
+    if (type === 'way') {
+      osm.log.get(version, function (err, node) {
+        if (err) return cb(err)
+        return cb(null, node.value.refs)
+      })
+    } else {
+      cb(null, [])
     }
   })
-  return res
+}
+
+// OsmDb, OsmVersion -> String
+function getTypeOfDeletedDoc (osm, version, cb) {
```

Comment:
agreed. let's keep it as a fallback, but store `v.type` and use that as default. We should think about how to have more robust type checking. Changing type with a `put` would cause all kinds of havoc and we don't have a check for that right now. It would involve a performance overhead so perhaps there should be a `opts.safe`? One option: how about a separate "sub-hyperlog" for each type?